controllers/discovery/initDiscoveryTopicData.js

```diff
@@ -42,11 +46,23 @@ const InitDiscoveryTopicData = async (req, res) => {
     });
     let suggestedTopics = topicWithCommonFollowerResult;
 
+    let totalData = await sequelize.query(totalDataQuery, {
+      type: QueryTypes.SELECT,
+      replacements: {
+        userId
+      },
+      raw: true
+    });
+    totalData = totalData?.[0]?.total_data || 0;
+
     return res.status(200).json({
       success: true,
       message: `Fetch discovery data success`,
       suggestedTopics,
-      page: page + 1
+      page: page + 1,
+      total_page: totalData > 0 && limit > 0 ? Math.ceil(totalData / limit) : 0,
+      limit: limit,
+      offset: page * limit
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling for this block of code could be improved. Currently, if an error occurs during the execution of `sequelize.query(totalDataQuery)`, it will be caught by the outer try-catch block and only a console log will be printed. It would be better to handle this error separately and send a response back to the client indicating that something went wrong.

```diff
-    let totalData = await sequelize.query(totalDataQuery, {
-      type: QueryTypes.SELECT,
-      replacements: {
-        userId
-      },
-      raw: true
-    });
-    totalData = totalData?.[0]?.total_data || 0;
+    let totalData;
+    try {
+      const result = await sequelize.query(totalDataQuery, {
+        type: QueryTypes.SELECT,
+        replacements: {
+          userId
+        },
+        raw: true
+      });
+      totalData = result?.[0]?.total_data || 0;
+    } catch (error) {
+      console.error(error);
+      return res.status(500).json({
+        success: false,
+        message: 'An error occurred while fetching total data',
+      });
+    }
```

Also, the `userId` is being passed as a replacement in the `totalDataQuery`, but it's not used in the query. If it's not needed, you should remove it to avoid confusion.

```diff
-      replacements: {
-        userId
-      },
+      replacements: {},
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->